deploy/deploy_test.go

```diff
@@ -720,6 +720,88 @@ func TestIncludeExclude(t *testing.T) {
 	}
 }
 
+// if we delete files locally that don't match the include/exclude pattern
+// the sync should not delete them on the remote.
+func TestIncludeExcludeRemoteDelete(t *testing.T) {
+	ctx := context.Background()
+
+	tests := []struct {
+		Include string
+		Exclude string
+		Want    deploySummary
+	}{
+		{
+			Want: deploySummary{NumLocal: 3, NumRemote: 5, NumUploads: 0, NumDeletes: 2},
+		},
+		{
+			Include: "**aaa",
+			Want:    deploySummary{NumLocal: 2, NumRemote: 3, NumUploads: 0, NumDeletes: 1},
+		},
+		{
+			Include: "aaa",
+			Want:    deploySummary{NumLocal: 1, NumRemote: 1, NumUploads: 0, NumDeletes: 0},
+		},
+		{
+			Exclude: "**bbb",
```

Comment:
I think this testcase only passes because the initial deploy is applying the filters; it's not really testing the filtering being applied to the remote.

You might need to do the first deploy with a target with no include/exclude filters, to make sure remote is set up with *all* the files, do the deletes, and then verify the appropriate deletes are skipped.

Reply:
You are right, one test wasn't actually uploading anything on the second deploy because nothing changed so it was testing the outcome of the first upload which wasn't useful.

I've changed the tests and made sure the initial deploy is just uploading with no include/exclude.